checker/src/org/checkerframework/checker/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,770 @@
+                String[] names = getIntersectingNames(a1, a2);
+
+                return createLTEqLengthOfAnnotation(names);
+            } else if ((AnnotationUtils.areSameByClass(a1, LTOMLengthOf.class)
+                            && AnnotationUtils.areSameByClass(a2, LTEqLengthOf.class))
+                    || (AnnotationUtils.areSameByClass(a1, LTEqLengthOf.class)
+                            && AnnotationUtils.areSameByClass(a2, LTOMLengthOf.class))) {
+
+                String[] names = getIntersectingNames(a1, a2);
+                return createLTEqLengthOfAnnotation(names);
+            } else if ((AnnotationUtils.areSameByClass(a1, LTLengthOf.class)
+                            && AnnotationUtils.areSameByClass(a2, LTOMLengthOf.class))
+                    || (AnnotationUtils.areSameByClass(a1, LTOMLengthOf.class)
+                            && AnnotationUtils.areSameByClass(a2, LTLengthOf.class))) {
+
+                String[] names = getIntersectingNames(a1, a2);
+                return createLTLengthOfAnnotation(names);
+            }
+            // Annotations are in this hierarchy, but they are not the same.
+            else {
+                return UNKNOWN;
+            }
+        }
+
+        /**
+         * Computes subtyping as per the subtyping in the qualifier hierarchy structure unless both
+         * annotations are the same. In this case, rhs is a subtype of lhs iff lhs contains at least
+         * every element of rhs.
+         *
+         * @return true if rhs is a subtype of lhs, false otherwise.
+         */
+        @Override
+        public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
+
+            if (AnnotationUtils.areSameByClass(lhs, UpperBoundUnknown.class)) {
+                return true;
+            } else if (AnnotationUtils.areSameByClass(rhs, UpperBoundUnknown.class)) {
+                return false;
+            } else if (AnnotationUtils.areSameByClass(rhs, UpperBoundBottom.class)) {
+                return true;
+            } else if (AnnotationUtils.areSameByClass(lhs, UpperBoundBottom.class)) {
+                return false;
+            } else if (AnnotationUtils.areSameIgnoringValues(lhs, rhs)) {
+                // Same type, so might be subtype.
+                List<Object> lhsValues = getElementValueArray(lhs, "value", Object.class, true);
+                List<Object> rhsValues = getElementValueArray(rhs, "value", Object.class, true);
+                return rhsValues.containsAll(lhsValues);
+            } else if (isSubtypeRelaxed(rhs, lhs)) {
+                /* Different types that are subtypes of each other ->
+                 * rhs is a subtype of lhs iff lhs.value contains rhs.value.
+                 */
+                List<Object> lhsValues = getElementValueArray(lhs, "value", Object.class, true);
+                List<Object> rhsValues = getElementValueArray(rhs, "value", Object.class, true);
+                return rhsValues.containsAll(lhsValues);
```

Comment:
I actually think it's the documentation that's wrong here; I'm going to fix that, but the intuition for why it's wrong is that you should always be able to assign a type with more information into a type with less. For instance, consider this:

```java
void test (@LTLengthOf({"a", "b"}) int iab, int[] a, int[] b) {
    @LTLengthOf("a") int ia = iab;
}
```

That assignment is definitely safe, and should be allowed by the type rules. The current rules do allow it, but switching lhs and rhs in isSubtype would cause it to fail.